### PR TITLE
[fix]: [PL-21074]: Disable grpcCall POLICYMGMT to reduce error logging

### PR DESCRIPTION
**Pull request by rathodmeetsatish**
Disable grpcCall for POLICYMGMT to reduce error logs

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31118)

### CODE REVIEW

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31118)* status: 3 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java, line 93 at r1](https://reviewable.io/reviews/harness/harness-core/31118#-MxIOBVaEYaC0Lvkr--A:-MxIOBVaEYaC0Lvkr--B:b-3w4v98) ([raw file](https://github.com/harness/harness-core/blob/72a0d5d94589a0cd794b67a964371a2a05195d7c/136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java#L93)):*
> ```Java
>       }
>       log.info("Saving {} files for the microservice {}", fileNumber, microservice);
>       if (entitiesForFullSync != null) {
> ```

This complete function is not looking clean. With every iteration we have added something in this function. Can we please make this function clean as a part of this ticket itself. To do so, we can break this function into multiple smaller parts.

**Review by rathodmeetsatish — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31118)* status: 3 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java, line 77 at r1](https://reviewable.io/reviews/harness/harness-core/31118#-Mx8_Es0fE9eSG_r-mww-r1-77:-MxQCA3J4JwUqqdQW6oP:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/72a0d5d94589a0cd794b67a964371a2a05195d7c/136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java#L77)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

lets make a method here -- if msvc has git sync feature and add this condition there to make logic clear and extendible
</blockquote></details>

Done.
___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java, line 93 at r1](https://reviewable.io/reviews/harness/harness-core/31118#-MxIOBVaEYaC0Lvkr--A:-MxQCCbV52zTKJlaYymq:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/72a0d5d94589a0cd794b67a964371a2a05195d7c/136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java#L93)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

This complete function is not looking clean. With every iteration we have added something in this function. Can we please make this function clean as a part of this ticket itself. To do so, we can break this function into multiple smaller parts.
</blockquote></details>

Done.

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java [hunk 1]
```diff
@@ -111,6 +103,31 @@ public void triggerFullSync(FullSyncEventRequest fullSyncEventRequest, String me
...
+  private boolean checkIfEntitiesAvailableForFullSync(FileChanges entitiesForFullSync, Microservice microservice) {
+    int fileNumber = emptyIfNull(entitiesForFullSync.getFileChangesList()).size();
+    if (fileNumber > 0) {
+      return true;
+    }
+    log.info("Saving {} files for the microservice {}", fileNumber, microservice);
+    return false;
+  }
+
+  private FileChanges makeGrpcCallIfFullSyncEnabled(Microservice microservice, String messageId,
```
Comment (abhinav-harness) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java:115]: better structure would be a method which says -- isFullSyncEnabled for msvc and does things. psuedo code
``` 
if(msvc=POLICY)
return false; 
return true 
```

**Review by rathodmeetsatish — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31118)* status: 3 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java, line 115 at r2](https://reviewable.io/reviews/harness/harness-core/31118#-Mx8_Es0fE9eSG_r-mww-r2-115:-MxXjO103tysmJJ3stJ7:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/cba4933df3e5568cab9a08e5db16ff9634984bce/136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java#L115)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

better structure would be a method which says -- isFullSyncEnabled for msvc and does things. psuedo code
``` 
if(msvc=POLICY)
return false; 
return true 
```
</blockquote></details>

Done.

**Comment by vistaarjuneja** [comment]
trigger ti0

**Review by bhavya181 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31118)* status: 3 file(s) to review. 2 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java, line 120 at r3](https://reviewable.io/reviews/harness/harness-core/31118#-MxYz4vcEVDEHRY7y_Df:-MxYz4vdBseQmHvyB4Rs:bza5btk) ([raw file](https://github.com/harness/harness-core/blob/a71b18ba05ddcb2b865b6b999f82cdc029bbca1f/136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java#L120)):*
> ```Java
>     int fileNumber = emptyIfNull(entitiesForFullSync.getFileChangesList()).size();
>     if (fileNumber > 0) {
>       return true;
> ```

Don't return from here, log will never be printed, just mark isEntitiesAvailableForFullSync as true here.

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java [hunk 2]
```diff
@@ -111,6 +113,20 @@ public void triggerFullSync(FullSyncEventRequest fullSyncEventRequest, String me
     }
   }
 
+  private boolean checkIfEntitiesAvailableForFullSync(FileChanges entitiesForFullSync, Microservice microservice) {
+    boolean isEntitiesAvailableForFullSync = false;
+    int fileNumber = emptyIfNull(entitiesForFullSync.getFileChangesList()).size();
```
Comment (bhavya181) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/FullSyncAccumulatorServiceImpl.java:118]: put a log here

**Review by bhavya181 — DISMISSED** [review]
> Reviewed 1 of 3 files at r1, 1 of 2 files at r2, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31118)* status: 2 file(s) to review. 1 change requests. 3. unresolved discussions. 1 of 2 approvals obtained.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31118)* status: 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by abhinav-harness — COMMENTED** [review]
> Reviewed 1 of 3 files at r1, 1 of 2 files at r2, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31118)* status: 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by abhinav-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31118)* status: :shipit: complete! ✓

**Pull request merged**